subsys/caf/modules/sensor_sampler.c

```diff
@@ -19,11 +20,29 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_SENSOR_SAMPLER_LOG_LEVEL);
 
 #define SAMPLE_THREAD_STACK_SIZE	CONFIG_CAF_SENSOR_SAMPLER_THREAD_STACK_SIZE
 #define SAMPLE_THREAD_PRIORITY		CONFIG_CAF_SENSOR_SAMPLER_THREAD_PRIORITY
+#define SAMPLE_BUF_SIZE			CONFIG_CAF_SENSOR_SAMPLER_DATA_BUF_SIZE
+
+static struct k_sem sample;
+static struct sensor_value buf[SAMPLE_BUF_SIZE];
```

Comment:
`buf` is not the right name. Why not making prev an array in sensor data structure instead of passing a pointer to this enigmatic buffer?

Reply:
Once application defines few sensors it will not work. The prev size would vary depending on sensor configuration. That's why i decided to 'allocate' this from common buffer.

Reply:
Used `k_malloc` instead as discussed offline.